src/cascadia/ShellExtension/OpenTerminalHere.h

```diff
@@ -52,9 +52,16 @@ struct
     STDMETHODIMP GetCanonicalName(GUID* pguidCommandName);
     STDMETHODIMP EnumSubCommands(IEnumExplorerCommand** ppEnum);
 #pragma endregion
+#pragma region IObjectWithSite
+    IFACEMETHODIMP SetSite(IUnknown* site) noexcept;
+    IFACEMETHODIMP GetSite(REFIID riid, void** site) noexcept;
+#pragma endregion
 
 private:
-    std::wstring _GetPathFromExplorer() const;
+    HRESULT GetLocationFromSite(IShellItem** location) const;
+    HRESULT GetBestLocationFromSelectionOrSite(IShellItemArray* psiArray, IShellItem** location) const;
+
+    winrt::com_ptr<IUnknown> site_;

```

Comment:
Could you replace `winrt::com_ptr` with `wil::com_ptr`? It's much more advanced, throws more detailed exceptions and has non-throwing variants for `as` (aka `query`), etc.

Reply:
I didn't want to blow up the existing code, but... it turns out that I mostly DID, so why not